internal/irc/handler.go

```diff
@@ -219,7 +223,17 @@ func (h *Handler) Run() error {
 			func() error {
 				h.log.Debug().Msgf("connect attempt %d", connectAttempts)
 
+				// #1239: don't retry if the user manually disconnected with Stop()
+				h.m.RLock()
+				defer h.m.RUnlock()
```

Comment:
I recommend against holding a mutex across a network operation like `Connect()`, even if it fixes a TOCTOU race